plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/utils/ReplicationControllerResourceProcessor.java

```diff
@@ -0,0 +1,102 @@
+import com.openshift.restclient.model.IResource;
+
+/**
+ * Processor for ReplicationControllerResourceProcessor. Scale ReplicationController to 0 and delete it.
+ * 
+ * @author Jeff Maury
+ *
+ */
+public class ReplicationControllerResourceProcessor extends BaseResourceProcessor {
+
+    public ReplicationControllerResourceProcessor() {
+        super(true);
+    }
+    
+    @Override
+    public void handleDelete(ConnectionsRegistry registry, Connection connection, IResource resource, boolean willDeleteSubResources)
+            throws OpenShiftException {
+        IReplicationController rc = (IReplicationController)resource;
+
+        try {
+            if (willDeleteSubResources) {
+                List<IResource>[] overlappingRCs = findOverlappingReplicationControllers(connection, rc);
+                if (!overlappingRCs[1].isEmpty()) {
+                    String list = overlappingRCs[1].stream().map(res -> res.getName()).collect(Collectors.joining(","));
+                    throw new OpenShiftException(
+                            "Found overlapping replication controllers for replication controller %s: %s", rc.getName(),
+                            list);
+                } else if (overlappingRCs[0].size() == 1) {
+                    rc.setDesiredReplicaCount(0);
+                    rc = connection.updateResource(rc);
+                    while (rc.getCurrentReplicaCount() > 0) {
+                        Thread.sleep(1000);
+                        rc = connection.getResource(rc);
+                    }
+                } 
+            }
+            super.handleDelete(registry, connection, resource, willDeleteSubResources);
+        } catch (InterruptedException e) {
+            throw new OpenShiftException(e, e.getLocalizedMessage());
+        }
+    }
+
+    private List<IResource>[] findOverlappingReplicationControllers(Connection connection, IReplicationController replicationController) {
+        List[] overlappingRCS = new List[2];
+        overlappingRCS[0] = new ArrayList<>();
+        overlappingRCS[1] = new ArrayList<>();
+        List<IResource> rcs = connection.getResources(ResourceKind.REPLICATION_CONTROLLER, replicationController.getNamespace());
+        for(IResource rc : rcs) {
+            if (match((IReplicationController)rc, replicationController)) {
+                if (((IReplicationController)rc).getReplicaSelector().size() == replicationController.getReplicaSelector().size()) {
+                    overlappingRCS[0].add(rc);
+                } else {
+                    overlappingRCS[1].add(rc);
+                }
+            }
+        }
+        return overlappingRCS;
+    }
+    
+    private boolean match(IReplicationController rc, IReplicationController replicationController) {
+        return match(rc.getReplicaSelector(), replicationController.getReplicaSelector()) |
+               match(replicationController.getReplicaSelector(), rc.getReplicaSelector());
```

Comment:
If map1.size() > map2.size() we can tell that map1 is not contained by map2 without calling match(map1,map2). So, below for implementation of match(Map,Map) I suggested switching maps so that map1.size() <= map2.size() and then just map2.entrySet().containsAll(map1.entrySet()) should do all the job.